_episodes/l1-03-intermediate_concepts.md

```diff
@@ -1,21 +1,441 @@
+> > to decide on its own how to remove the onion given that something has been added in the
+> > same part of the recipe afterwards.
+> >
+> > The ingredients file now will look like this:
+> > ~~~
+> > * 2 avocados
+> > * 1 lime
+> > * 2 tsp salt
+> > <<<<<<< HEAD
+> > * 1/2 onion
+> > * 1 tbsp coriander
+> > =======
+> > >>>>>>> parent of 5cb4883 (Added 1/2 onion)
+> > ~~~
+> > {: .output}
+> >
+> > To move forward, fix the conflicts as it was done in the previous section - removing the
+> > << and >> lines as well as "1/2 onion" and run:
+> > ~~~
+> > $ git add ingredients.md
+> > $ git revert --continue --no-edit
+> > $ git graph
+> > ~~~
+> > {: .commands}
+> > ~~~
+> > * 53371e5 (HEAD -> main) Revert "Added 1/2 onion"
+> > *   fe0d257 Merge branch 'experiment'
+> > |\
+> > | * 99b2352 Reduced the amount of coriander
+> > * | 2c2d0e2 Merge branch 'experiment'
+> > |\|
+> > | * d9043d2 Try with some coriander
+> > * | 6a2a76f Corrected typo in ingredients.md
+> > |/
+> > * 57d4505 Revert "Added instruction to enjoy"
+> > * 5cb4883 Added 1/2 onion
+> > * 43536f3 Added instruction to enjoy
+> > * 745fb8b Adding ingredients and instructions
+> > ~~~
+> > {: .output}
+> > Using `git revert` has added a new commit which reverses *exactly* the changes made in
+> > the specified commit (after solving the conflict).
+> >
+> {: .solution}
+>
+{: .challenge}
+
+This is yet another good example of why making separate commits for each change is a
+good idea, so they can, potentially, be reversed if needed in the future with no fuss.
+
+### Set aside your work safely with `stash`
+
+It is not rare that, while you are working on some feature, you need to check something
+else in another branch. Very often this is the case when you want to try some
+contributor's code as part of a pull request review process (see next episodes). You
+can commit the work you are doing, but if it is not in a state ready to be committed,
+what would you do?
+
+`git stash` is the answer. It lets you put your current, not committed work aside in a
+special state, reverting the working directory to the way it was in the last commit.
+Then, you can easily switch branches, pull new ones or do whatever you want. Once you
+are ready to go back to work, you can recover the stashed work as continue as if
+nothing had happen.
+
+The following are the `git stash` commands needed to make this happen:
+
+- Stash the current state of the repository, giving some message to remind yourself what
+  was this about. The working directory becomes identical to the last commit.
+~~~
+$ git stash save "Some informative message"
+~~~
+{: .commands}
+- List the stashes available in reverse chronological order (last one stashed goes on
+  top).
+~~~
+$ git stash list
+~~~
+{: .commands}
+- Extracts the stash with the given number from the list, updating the working directory
+  with its content.
+~~~
+$ git stash pop stash@{NUMBER}
```

Comment:
I think we should also at least mention that if you run `git stash pop` without an extra arg it just pops the last stash as that's the usual way of using the command. Ditto below for `git stash apply`.